### ISSUE TITLE
Unsupported protocol scheme error when trying to use https 

### ISSUE THREAD

**Issue by pellegrino** [dnaeon/go-vcr#1]
Thanks for the great work with go-vcr - it is exactly what I was looking for. I'm trying to understand how to solve this issue but my still limited go skills are getting in the way so I figured out it would be helpful to create a ticket. 

```go 
	r, err := recorder.New("../../testdata/githubapi")
	if err != nil {
		t.Fatal(err)
	}
	defer r.Stop() // Make sure recorder is stopped once done with it

	// Create an HTTP client and inject our transport
	client := &http.Client{
		Transport: r.Transport, // Inject our transport!
	}

	response, err := client.Get("https://example.com")
```
When trying to use a https URL, go-vcr will fail complaining that the protocol scheme is unsupported and it will fail with:  

```
2015/12/29 09:42:11 Failed to process request for URL //example.com:443: unsupported protocol scheme ""
```

I will keep looking and open a PR in case I find a solution. 

Cheers,

**Comment by dnaeon** [comment]
Hey @pellegrino,

Sorry for the late response, but I've been away for some time and had limited access to my computer.

Currently `go-vcr` does not handle HTTPS URLs, because we are not handling the `CONNECT` HTTP method when a new HTTPS request is being sent over the `go-vcr` transport/proxy.

We should have a special case if `r.Method == "CONNECT"` and handle it in the `requestHandler` from https://github.com/dnaeon/go-vcr/blob/master/recorder/recorder.go#L60 

I will not be able to look into this one soon, as I will not be near a computer for the next couple of weeks, but if you have time to hack on this meanwhile it would be great!

You can find more info about `CONNECT` in the links below as well if you decide to give it a shot:

* https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods
* https://en.wikipedia.org/wiki/HTTP_tunnel#HTTP_CONNECT_tunneling
* https://www.ietf.org/rfc/rfc2817

Thanks,
Marin

**Comment by dnaeon** [comment]
Fixed #6
